src/FwdState.cc

```diff
@@ -801,14 +804,22 @@ FwdState::noteConnection(HappyConnOpener::Answer &answer)
         Must(!Comm::IsConnOpen(answer.conn));
         answer.error.clear(); // preserve error for errorSendComplete()
     } else if (!Comm::IsConnOpen(answer.conn) || fd_table[answer.conn->fd].closing()) {
+        // We do not know exactly why the connection got closed, so we play it
+        // safe: Do not ban retries (i.e. do not set flags.dont_retry) but do
+        // not retry this potentially problematic destination either (i.e. do
+        // not set destinationReceipt).
```

Comment:
However, we could check 'answer.reused' and set destinationReceipt if it is true. Should not we do that?

Reply:
OK, please refactor to do that.